### PR TITLE
Admin merchant dashboard #91

### LINKED ISSUE #91: User Story 43, Admin can downgrade a merchant to regular user
```
As an admin user
When I visit a merchant's dashboard ("/admin/merchants/6")
I see a link to "downgrade" the merchant's account to become a regular user
The merchant themselves do NOT see this link
When I click on that link
I am redirected to ("/admin/users/6") because the merchant is now a regular user
And I see a flash message indicating the user has been downgraded
The next time this user logs in they are no longer a merchant
Only admins can see the "downgrade" button
Only admins can reach any route necessary to downgrade the merchant to user status
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

### PR DESCRIPTION
**Pull request by nicovigil1**
closes #91 adds ability for admin to toggle a merchant's role on the admin_merchant_path

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -52,6 +52,15 @@ def switch_enabled
...
 
+    def toggle_role
+      if self.role == "default"
+        self.role = "merchant"
+      elsif self.role == "merchant"
+        self.role = "default"
+      end 
+      save
+    end 
+
```
Comment (stoic-plus) [inline app/models/user.rb]: What I would do here nico:

```
self.role = self.role == "default" ? "merchant" : "default"
```

This is called a _ternary_ operator. `self.role` will be assigned the first one ("merchant") if the condition evaluates to true and the second one if false.

Comment (stoic-plus) [reply app/models/user.rb]: `self.role == "default" ?`
That is the condition part

File: app/views/merchants/index.html.erb [hunk 2]
```diff
@@ -55,15 +55,32 @@
...
+              <% end %>
+              <td><%= m.city %></td>
+              <td><%= m.state %></td>
+              <td><%= m.created_at.to_date %></td>
+            </tr>
+        <% end %>
+      <% end %>
+    </tbody>
+  </table>     
+</div>
```
Comment (stoic-plus) [inline app/views/merchants/index.html.erb]: This looks clean and concise. Great job Nico!

File: app/models/user.rb (hunk 1 above)
Comment (nicovigil1) [reply app/models/user.rb]: btw I'm 2 branches ahead so I'm just gonna add this change on my most recent branch

Comment (nicovigil1) [reply app/models/user.rb]: btw I'm 2 branches ahead so I'm just gonna add this change on my most recent branch

File: app/views/merchants/index.html.erb [hunk 3]
```diff
@@ -55,15 +55,39 @@
...
+                <td><%= link_to m.name, admin_merchant_path(m) %></td>
+              <% else %>
+                <td><%= m.name %></td>
+              <% end %>
+              <td><%= m.city %></td>
+              <td><%= m.state %></td>
+              <td><%= m.created_at.to_date %></td>
+              <% if current_user && current_user.role == 'admin' %>
+                <% if m.enabled? %>
+                  <td><%= button_to 'disable', admin_toggle_path, params:{user_id:m.id} %></td>
```
[the remark this answers is not in the document]
Comment (nicovigil1) [reply app/views/merchants/index.html.erb]: good catch!
